File: src/main/java/com/example/Imdb_Clone/service/WatchlistService.java
package com.example.Imdb_Clone.service;

import com.example.Imdb_Clone.dto.MovieResponseDto;

import java.util.Set;

public interface WatchlistService {
    void addMovieToWatchlist(String username, Long movieId);

    void removeMovieFromWatchlist(String username, Long movieId);

    Set<MovieResponseDto> getWatchlist(String username);
}

File: frontend/imdb-clone-frontend/src/services/watchlistService.js
import apiClient from './apiService';

const getWatchlist = () => {
    return apiClient.get('/watchlist');
};

const addMovieToWatchlist = (movieId) => {
    return apiClient.post(`/watchlist/movies/${movieId}`);
};

const removeMovieFromWatchlist = (movieId) => {
    return apiClient.delete(`/watchlist/movies/${movieId}`);
};

const watchlistService = {
    getWatchlist,
    addMovieToWatchlist,
    removeMovieFromWatchlist,
};

export default watchlistService;
